daras_ai_v2/base.py

```diff
@@ -325,6 +315,17 @@ def sentry_event_set_user(self, event, hint):
             }
         return event
 
+    def _load_state_from_query_params(self):
+        request_state = self.request.query_params.get(REQUEST_STATE_Q)
+        if not request_state:
+            return
+
+        try:
+            request_state = base64.urlsafe_b64decode(request_state).decode()
```

Comment:
but for anon users, we won't have their real `uid` until they login.

are you suggesting that we create a run without a `uid`, and when the user comes back with `SUBMIT_AFTER_LOGIN_Q`, we then update the `uid` and run?

Reply:
we generate a firebase user as soon as you submit, this should do the same thing

Reply:
for anon users who come back with an existing account, we would have 2 different uids.

i fixed this case by setting the uid of the unqueued run when user comes back. i added a concurrency check there too so that we don't queue the same run twice.